synapse/exc.py

```diff
@@ -77,6 +77,9 @@ class BadCoreStore(SynErr):
     '''The storage layer has encountered an error'''
     pass
 
+class CantDelProp(SynErr): pass
+class CantSetProp(SynErr): pass
```

Comment:
I think this will be difficult to do until https://github.com/vertexproject/synapse/issues/359 is complete